packages/shorebird_cli/lib/src/commands/release_new/android_releaser.dart

```diff
@@ -0,0 +1,205 @@
+
+/// {@template android_releaser}
+/// Functions to create an Android release.
+/// {@endtemplate}
+class AndroidReleaser extends Releaser {
+  /// {@macro android_releaser}
+  AndroidReleaser({
+    required super.argResults,
+    required super.flavor,
+    required super.target,
+  });
+
+  @override
+  ReleaseType get releaseType => ReleaseType.android;
+
+  Set<Arch> get architectures => (argResults['target-platform'] as List<String>)
+      .map(
+        (platform) => AndroidArch.availableAndroidArchs
+            .firstWhere((arch) => arch.targetPlatformCliArg == platform),
+      )
+      .toSet();
+
+  late bool generateApk = argResults['android-artifact'] as String == 'apk';
+  late bool splitApk = argResults['split-per-abi'] == true;
+
+  @override
+  Future<void> validatePreconditions() async {
+    try {
+      await shorebirdValidator.validatePreconditions(
+        checkUserIsAuthenticated: true,
+        checkShorebirdInitialized: true,
+        validators: doctor.androidCommandValidators,
+      );
+    } on PreconditionFailedException catch (e) {
+      exit(e.exitCode.code);
+    }
+  }
+
+  @override
+  Future<void> validateArgs() async {
+    if (generateApk && splitApk) {
+      logger
+        ..err(
+          'Shorebird does not support the split-per-abi option at this time',
+        )
+        ..info(
+          '''
+Split APKs are each given a different release version than what is specified in the pubspec.yaml.
+
+See ${link(uri: Uri.parse('https://github.com/flutter/flutter/issues/39817'))} for more information about this issue.
+Please comment and upvote ${link(uri: Uri.parse('https://github.com/shorebirdtech/shorebird/issues/1141'))} if you would like shorebird to support this.''',
+        );
+      exit(ExitCode.unavailable.code);
+    }
+  }
+
+  @override
+  Future<FileSystemEntity> buildReleaseArtifacts() async {
+    final architectures = (argResults['target-platform'] as List<String>)
+        .map(
+          (platform) => AndroidArch.availableAndroidArchs
+              .firstWhere((arch) => arch.targetPlatformCliArg == platform),
+        )
+        .toSet();
+
+    final flutterVersionString = await shorebirdFlutter.getVersionAndRevision();
+
+    final buildAppBundleProgress = logger
+        .progress('Building app bundle with Flutter $flutterVersionString');
+    await artifactBuilder.buildAppBundle(
+      flavor: flavor,
+      target: target,
+      targetPlatforms: architectures,
+    );
+    buildAppBundleProgress.complete();
+
+    if (generateApk) {
+      final buildApkProgress =
+          logger.progress('Building APK with Flutter $flutterVersionString');
+      await artifactBuilder.buildApk(
+        flavor: flavor,
+        target: target,
+        targetPlatforms: architectures,
+      );
+      buildApkProgress.complete();
+    }
+
+    final projectRoot = shorebirdEnv.getShorebirdProjectRoot()!;
+    try {
+      return shorebirdAndroidArtifacts.findAab(
```

Comment:
Both the `buildAppbundle` and `buildApk` do the `find*` before returning, and throws the exceptions.

So I guess you don't need this additional `findAab` here, and the `try/catch` should wrap the whole method

Reply:
Fixed